### ISSUE TITLE
Installation media isn't readonly

### ISSUE THREAD

**Issue by KloudKoder** [pop-os/pop#1144]
**Distribution (run `cat /etc/os-release`):**
 
20.04 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

ISO image

**Issue/Bug Description:**

Somehow, I assume from the ISO but maybe later in the install process, there arises a writable partition on my USB drive used for the installation. This is a security problem because it allows information from one install to propagate to the next, and the next, etc. I trust that you're not so dumb as to dump unencrypted passwords onto on. On the other hand, I don't know _what_
you're dumping to it, so it's a problem. Fundamentally, I can't even verify that evil firmware on machine A didn't jump onto the USB stick in order to infect Machine B, because the hash of a dd is going to vary from install to install. Badness. And finally, why is this necessary? Presumably, I could dust off the old DVD burner and install from that readonly media. Can I now not do that? Is the writable partitiion just used to improve performance at the aforementioned expense? Definitely not worth it.

**Steps to reproduce (if you know):**

"sudo dd" the Pop!_OS 20.04 ISO onto a USB stick. Then, after it finishes writing, eject it. Reinsert it. "sudo dd" it back to a file. You can stop the dd process when the file size exceeds the ISO size. Truncate the file to the same exact size as the ISO. Take its sha256sum. Verify that it matches the ISO's. Now use it to do an install. Notice that the sha256sum changes. On closer inspection, this appears to be due to the presence of a writable partition. This hides any malicious changes that the installed-to machine may have made to sensitive areas.

**Expected behavior:**

Don't change one bit on the install media.

**Other Notes:**

I don't know any other distro that behaves like this. Maybe this is yet another horror to come out of Ubuntu. In any case, it's bad.

**Comment by crawfxrd** [comment]
The installation partition is readonly. There is a writable partition that records installation logs.

**Comment by KloudKoder** [comment]
@crawfxrd That's true, but that makes it tough to verify because one needs to know where the readonly partition ends and the writable one begins; you can't just read the entire USB stick and truncate it to the size of the Pop!_OS ISO. One compromise would be to overwrite the writable partition back to its initial state (like all 0s or something) before the install does its first reboot. I guess one could also verify only the readonly partition (like /dev/sdc0 instead of /dev/sdc, for example), but this wouldn't include the very critical partition table at the beginning of the media.
